### PR TITLE
overhaul CI

### LINKED ISSUE #16: examples is a submodule?
The CI in #10 was meant to update to `examples/`, but seems to be treating it as a submodule. This should be fixed so that `examples/` ideally contains its own submodules, or just example directories.

### LINKED ISSUE #23: Get last minor release of Python in CI
Currently we're using 3.9 for general tests in the cookie CI. This will become outdated as more versions of Python get released. It would be handy to automatically get the last minor release of Python instead.

More discussion here: https://github.com/MDAnalysis/cookiecutter-mdakit/pull/7#discussion_r928771324

I think this could be doable by just inspecting which versions of Python are available in the `setup-python` actions, since that's the limiting factor for setting up Python anyway.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #23 
Fixes #16 

This PR redoes CI, which now:

* runs the cookie CI in a step that will fail if the cookie CI fails (only ubuntu and with conda-forge, for reasons like https://github.com/nektos/act/issues/561)
* Uploads artifacts to inspect
* On merge to main, pushes example repositories to https://github.com/MDAnalysis/mdakit-cookie/ under various branches. Here, CI will also run, but won't be a failing step in the cookiecutter repo. (Sorry for eating up all our private budget for this month :( )
* On merge to main, pushes example repositories to the `examples/` branch of this one

Despite ostensibly addressing fairly low-priority issues, this fixes CI so that it actually runs -- turns out the cookie tests weren't actually working as in the `main` branch. Therefore it's necessary.

Decisions that could do with discussion:
* the cookie action `get-stable-python-version` is reasonably re-usable, if we want to store it somewhere in the MDAnalysis/ repo and just pull from that. It's licensed under MIT and afaik I wrote it from scratch -- are we ok with that?
* I don't know if we have a quota for uploading artifacts. I set the retention time to 3 days (down from the default 90) just in case.

### CODE REVIEW

**Comment by lilyminium** [comment]
@orbeckst @IAlibay @fiona-naughton your reviews and opinions on the items in the first comment would be highly appreciated, but I do realise that this is a fairly large PR. If one of you is happy with the behaviour I think it adds, and with the examples at https://github.com/MDAnalysis/cookiecutter-mdakit/tree/examples and https://github.com/MDAnalysis/mdakit-cookie branches, then I'll count that as good enough :)

File: {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+)
+parser.add_argument(
+    "--field",
+    choices=FIELD_CHOICES,
+    help="Field to select last-n-version on",
+    default="minor",
+)
+
+
+class VersionInfo(NamedTuple):
```
Comment (IAlibay) [inline {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py:76]: Isn't there a whole version package maintained by pypa that does this?

**Review by IAlibay — APPROVED** [review]
Very quick review, I'll try to give it another eye in a few hours when I'm more awake.

File: {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py (hunk 1 above)
Comment (lilyminium) [reply {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py:76]: Is it in the standard library? I couldn't find anything from a quick skim 😅 the script needs to be able to run on the base Python installs on the GH runners

Comment (IAlibay) [reply {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py:76]: maybe we can vendor version.py?

**Comment by orbeckst** [comment]
> I don't know if we have a quota for uploading artifacts. I set the retention time to 3 days (down from the default 90) just in case.

Yes, AFAIK there's a 500 GB quota on artifacts (I saw it when I set up the artifact uploading for the whitepaper). Reducing storage duration to 3d is a good idea.

**Comment by orbeckst** [comment]
I am in favor of putting re-useable actions into separate repos. MIT license is also totally fine with me.

**Review by orbeckst — APPROVED** [review]
My head is spinning a little bit from the whole CI within CI (?) and the plethora of things that are tested and created like https://github.com/MDAnalysis/cookiecutter-mdakit/tree/examples . 

Could you add some documentation to https://github.com/MDAnalysis/mdakit-cookie/tree/main (README on main, repo description) to indicate what the repo is good for? My understanding is that it is a testing/staging ground for the cookiecutter — ?

I am happy with MIT license everywhere here and making actions more easily reuseable by putting them in their own repos.

I didn't have time to dig in the details of the PR but I am happy with the big picture and so I add my +1.

File: {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py (hunk 1 above)
Comment (IAlibay) [reply {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py:76]: (my viewpoint here being that maintaining a fully compliant version string reader is quite the pain iirc and `packaging`'s version.py is sorta made to be vendored for these purposes iirc)

Comment (IAlibay) [reply {{cookiecutter.repo_name}}/.github/actions/get-stable-python-version/get_python_version.py:76]: actually maybe just ignore me here, that'd be a lot more complexity for this task

**Comment by IAlibay** [comment]
> > I don't know if we have a quota for uploading artifacts. I set the retention time to 3 days (down from the default 90) just in case.
> 
> Yes, AFAIK there's a ~500 GB~ 500 MB quota on artifacts (I saw it when I set up the artifact uploading for the whitepaper). Reducing storage duration to 3d is a good idea.

We actually burn through this real quick as an organisation, especially when we do releases (wheels for everything is a good 200 MB being used for a few days). I'd suggest going for the minimum artifact retention period that is useful here (even down to a single day unless you expect to need to inspect / debug artifact contents)

**Comment by orbeckst** [comment]
FYI,  https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions 


> Product | Storage | Minutes (per month) 
> -- | -- | --
> GitHub Free for organizations | 500 MB |  2,000

**Comment by orbeckst** [comment]
Maybe storage is actually free for public repositories and the limits only apply to private repos. I am not 100% sure

> GitHub Actions usage is free for both public repositories and self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage, depending on the product used with the account. 

In any case, keeping retention times short is good stewardship of resources.

**Comment by lilyminium** [comment]
Thanks for looking this over, @IAlibay  and @orbeckst! I made #27 as an issue so I can merge this CI now. I'll also keep an eye on the storage quota if the retention time becomes an issue. :)

**Pull request merged**